python/ray/tune/ray_trial_executor.py

```diff
@@ -285,17 +293,17 @@ def _setup_remote_runner(self, trial):
                     "implemented and return True.")
             return existing_runner
 
-        if self._cached_actor_pg[0]:
-            logger.debug("Cannot reuse cached runner {} for new trial".format(
```

Comment:
Nooby question, when we reuse placement group, do we cross check the new trial's resources requirement matches the placement group's?

Reply:
In `tune/utils/placement_group` this would throw an error:

```
        assert pgf == trial_pgf, f"Cannot assign placement group with a " \
                                 f"non-matching factory to trial {trial}"
```

so we effectively require users to only use `reuse_actors` for homogeneous trial resource requirements.

This is something we should improve in the future.